exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,241 @@ const (
+	// Name is the name of the CIDR Block.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// CidrBlock is the list of CIDR blocks associated with Authorized Network.
+	// +optional
+	CidrBlock map[string]string `json:"cidrBlock,omitempty"`
+}
+
+// AuthorizedNetwork provide an IP-based firewall that controls access to the GKE control plane.
+type AuthorizedNetwork struct {
+	// Enabled defines Whether enable authorized network or not.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// CidrBlocks defines the CIDR block configuration.
+	// +optional
+	CidrBlocks AuthorizedNetworkCidrBlock `json:"cidrBlocks,omitempty"`
+}
+
+// ClusterNetworkPod the range of CIDRBlock list from where it gets the IP address.
+type ClusterNetworkPod struct {
+	// CidrBlock is where all pods in the cluster are assigned an IP address from this range. Enter a range
+	// (in CIDR notation) within a network range, a mask, or leave this field blank to use a default range.
+	// This setting is permanent.
+	// +optional
+	CidrBlock string `json:"cidrBlock,omitempty"`
+}
+
+// ClusterNetworkService defines the range of CIDRBlock list from where it gets the IP address.
+type ClusterNetworkService struct {
+	// CidrBlock is where cluster services will be assigned an IP address from this IP address range. Enter a range
+	// (in CIDR notation) within a network range, a mask, or leave this field blank to use a default range.
+	// This setting is permanent.
+	// +optional
+	CidrBlock string `json:"cidrBlock,omitempty"`
+}
+
+// ClusterNetwork define the cluster network.
+type ClusterNetwork struct {
+	// PrivateCluster defines the private cluster spec.
+	// +optional
+	PrivateCluster PrivateCluster `json:"privateCluster,omitempty"`
+
+	// UseIPAliases is whether alias IPs will be used for pod IPs in the cluster. If false, routes will be used for
+	// pod IPs in the cluster.
+	// +optional
+	UseIPAliases bool `json:"useIPAliases,omitempty"`
+
+	// AuthorizedNetwork provide an IP-based firewall that controls access to the GKE control plane.
+	// +optional
+	AuthorizedNetwork AuthorizedNetwork `json:"authorizedNetwork,omitempty"`
+
+	// Pod defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Pod ClusterNetworkPod `json:"pod,omitempty"`
+
+	// Service defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Service ClusterNetworkService `json:"service,omitempty"`
+}
+
+// WorkloadIdentityConfig allows workloads in your GKE clusters to impersonate Identity and Access Management (IAM)
+// service accounts to access Google Cloud services.
+type WorkloadIdentityConfig struct {
+	// Enable is to enable the workload identity config.
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+
+	// WorkloadPool is the workload pool to attach all Kubernetes service accounts to Google Cloud services.
+	// Only relevant when enabled is true
+	// +optional
+	WorkloadPool string `json:"workloadPool,omitempty"`
```

Comment:
Assuming that this is required if we want to use workload identity. 

As general comment, if a struct is used to represent an optional feature like workload identity then we can make fields required as the parent is optional.